RetailCoder.VBE/Refactorings/ImplementInterface/ImplementInterfaceRefactoring.cs

```diff
@@ -175,23 +173,24 @@ private string PropertySetStmt(Declaration member)
 
         private List<Parameter> GetParameters(Declaration member)
         {
+            var parameters1 = _declarations.Where(item => item.DeclarationType == DeclarationType.Parameter &&
+                                                         item.ParentScope == member.Scope);
             var parameters = _declarations.Where(item => item.DeclarationType == DeclarationType.Parameter &&
-                              item.ParentScope == member.Scope)
+                              item.ParentScopeDeclaration == member)
                            .OrderBy(o => o.Selection.StartLine)
                            .ThenBy(t => t.Selection.StartColumn)
                            .Select(p => new Parameter
                            {
-                               Accessibility = ((VBAParser.ArgContext)p.Context).BYREF() == null ? Tokens.ByVal : Tokens.ByRef,
+                               Accessibility = ((VBAParser.ArgContext)p.Context).BYREF() != null || 
+                                        (((VBAParser.ArgContext)p.Context).BYREF() == null && ((VBAParser.ArgContext)p.Context).BYVAL() == null) 
+                                            ? Tokens.ByRef 
+                                            : Tokens.ByVal,
+
```

Comment:
Boy, that is unnecessarily ugly:

    Accessibility = ((VBAParser.ArgContext)p.Context).BYVAL() != null
        ? Tokens.ByVal 
         : Tokens.ByRef,